v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/FailedPubsubMessageToPubsubTopicFn.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.elasticsearch.transforms;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.cloud.teleport.v2.elasticsearch.utils.ElasticsearchUtils;
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import java.nio.charset.StandardCharsets;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.DoFn;
+
+/**
+ * The {@link FailedPubsubMessageToPubsubTopicFn} converts PubSub message which have failed processing into
+ * {@link com.google.api.services.pubsub.model.PubsubMessage} objects which can be output to a PubSub topic.
+ */
+public class FailedPubsubMessageToPubsubTopicFn
+        extends DoFn<FailsafeElement<PubsubMessage, String>, PubsubMessage> {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
```

Comment:
Better to add a counter to track failed messages
```
 /** Counter to track total failed messages. */
    private static final Counter ERROR_MESSAGES_COUNTER =
        Metrics.counter(FailedStringToPubsubMessageFn.class, "total-failed-messages");
```
Refer https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/0ab42b80ce6fbe46e8325195831e0a99fbf18adf/src/main/java/com/google/cloud/teleport/templates/common/ErrorConverters.java#L372